bigquery_etl/glean_usage/common.py

```diff
@@ -113,88 +141,120 @@ def _extract_dataset_from_glob(pattern):
-    )
-    parser.add_argument(
-        "--date",
-        required=True,
-        type=lambda d: datetime.strptime(d, "%Y-%m-%d").date(),
-        help="Date partition to process, in format 2019-01-01",
-    )
-    parser.add_argument(
-        "--output_dir",
-        "--output-dir",
-        help="Also write the query text underneath the given sql dir",
-    )
-    parser.add_argument(
-        "--output_only",
-        "--output-only",
-        "--views_only",  # Deprecated name
-        "--views-only",  # Deprecated name
-        action="store_true",
-        help=(
-            "If set, we only write out sql to --output-dir and we skip"
-            " running the queries"
-        ),
-    )
-    standard_args.add_parallelism(parser)
-    standard_args.add_dry_run(parser, debug_log_queries=False)
-    standard_args.add_log_level(parser)
-    standard_args.add_priority(parser)
-    standard_args.add_billing_projects(parser)
-    standard_args.add_table_filter(parser)
-    return parser
+def get_app_info():
+    """Return a list of applications from the probeinfo API."""
+    resp = requests.get(APP_LISTINGS_URL)
+    resp.raise_for_status()
+    apps_json = resp.json()
+    app_info = {}
 
+    for app in apps_json:
+        if app["app_name"] not in app_info:
+            app_info[app["app_name"]] = [app]
+        else:
+            app_info[app["app_name"]].append(app)
 
-def generate_and_run_query(run_query_callback, description):
-    """Generate and run queries using a threadpool.
+    return app_info
 
-    The callback function is reponsible for generating and running the queries.
-    This was the original main entrypoint for each of the usage queries.
-    """
-    parser = get_argument_parser(description)
-    args = parser.parse_args()
-
-    try:
-        logging.basicConfig(level=args.log_level, format="%(levelname)s %(message)s")
-    except ValueError as e:
-        parser.error(f"argument --log-level: {e}")
-
-    baseline_tables = list_baseline_tables(
-        project_id=args.project_id,
-        only_tables=getattr(args, "only_tables", None),
-        table_filter=args.table_filter,
-    )
 
-    with ThreadPool(args.parallelism) as pool:
-        # Do a first pass with dry_run=True so we don't end up with a partial success;
-        # we also write out queries in this pass if so configured.
-        pool.map(
-            partial(
-                run_query_callback,
-                args.project_id,
-                date=args.date,
-                dry_run=True,
-                output_dir=args.output_dir,
-                output_only=args.output_only,
-            ),
-            baseline_tables,
+class GleanTable:
+    """Represents a generated Glean table."""
+
+    def __init__(self):
+        """Init Glean table."""
+        self.target_table_id = ""
+        self.prefix = ""
+        self.custom_render_kwargs = {}
+        self.no_init = True
+
+    def generate_per_app_id(
```

Comment:
Most of the old `run_query` logic was pretty redundant. So I moved the logic into a `GleanTable` class (maybe there is a better name?) that the specific baseline table implementations can extend with some table specific configuration or logic.